docs/zkapps/advanced-snarkyjs/time-locked-accounts.mdx

```diff
@@ -0,0 +1,114 @@
+
+This graph shows how each of the timing properties affect the vesting schedule:
+
+<figure>
+  <img
+    src="/img/10_time_locked_accounts.png"
+    width="95%"
+    alt="Timing parameters"
+  />
+</figure>
+
+The red cross on the left marks the point in time where the `timing` field is set and `isTimed` switches from `false` to `true`. The blue line shows how the amount of unlocked tokens increases over time, until it finally reaches its maximum value and stays flat. At this point, `isTimed` flips from `true` back to `false`, because no tokens remain locked.
+
+As you can see, the maximum amount of unlocked tokens is defined by the `initialMinimumBalance`. It is called "initialMinimumBalance" because, even though the tokens show up in the balance, they can't be withdrawn — in other words, the account has a a non-zero _minimum balance_. Initially, that minimum balance is equal to the amount of tokens locked (= initial minimum balance). Over time, the minimum balance decreases until it hits zero, which is the condition that makes `isTimed` false again.
+
+Here's a breakdown of all timing fields:
+
+- **cliff time**: the initial time length during which no tokens whatsoever are unlocked.
+  - note: "time" is measured in slots (in Mina, currently 1 slot = 3 minutes).
+- **cliff amount**: the amount of tokens unlocked after the cliff time ended.
+  - if this is greater or equal the initial minimum balance, _all_ tokens get unlocked after the cliff time.
+- **vesting period**: after the cliff time, tokens get unlocked periodically in a fixed interval, by a fixed amount. The vesting period is the length of that interval.
+- **vesting increment**: the amount by which tokens get unlocked after each vesting period
+
+It's important to note that there is only ever one vesting schedule per account and it can't be changed during the vesting period. So, whenever `isTimed` is true, all the timing fields are immutable. When all tokens are unlocked and `isTimed` flips back to `false`, the account timing becomes mutable again.
+
+### Setting timing in SnarkyJS
+
+In SnarkyJS, `timing` is one of the account fields that can be updated by an account update:
+
+```ts
+accountUpdate.account.timing.set({ initialMinimumBalance, cliffTime, ...etc });
+```
+
+You need to set all timing parameters except `isTimed`, which is managed by the protocol automatically.
+
+Let's see how to correctly implement some simple example use cases.
+
+**Use case 1: all tokens unlocked after 1 week**. If you don't care about the linear vesting schedule and just want all tokens unlocked after a certain time, then the only parameters you really care about are `cliffTime` and `cliffAmount`. We also want to set `initialMinimumBalance` equal to `cliffAmount` to ensure all tokens are unlocked at this point. The values `vestingPeriod` and `vestingIncrement` do not matter so we set them to 1 and 0, respectively:
```

Comment:
> #### Example 1: All tokens unlock after 1 week
> 
> If you simply want all tokens to unlock after a certain time, then the only properties you need to care about are `cliffTime` and `cliffAmount`. We also want to set `initialMinimumBalance` equal to `cliffAmount` to ensure all tokens are unlocked when the cliff elapses. We can use a `vestingPeriod` of `1` and `vestingIncrement` is unused so we'll set it to `0`:

(probably should be `####`, but see how it looks)

Wouldn't it be a clearer UX to tell them they don't need to specify `vestingPeriod` and `vestingIncrement`, and then let SnarkyJS handle defaulting the values of these?

Reply:
yeah I agree @ UX

Reply:
added issue to improve this: https://github.com/o1-labs/snarkyjs/issues/770